package org.opentmf.dnext.tmf666.model;

import com.fasterxml.jackson.annotation.JsonTypeInfo;
import jakarta.validation.Valid;
import lombok.Getter;
import lombok.Setter;
import org.opentmf.dnext.common.model.QuoteInformationRequiredEventBase;
import org.opentmf.tmf666.model.ISettlementAccountAttributeValueChangeEvent;

/**
 * The notification data structure.
 *
 * <p><br/>
 * <strong>Referring TMF artifacts:</strong>
 * <ul>
 *   <li>TMF-666: Account Management API</li>
 * </ul>
 * </p>
 *
 * @author Gökhan Demir
 */
@Getter
@Setter
@JsonTypeInfo(
    use = JsonTypeInfo.Id.NAME,
    visible = true,
    include = JsonTypeInfo.As.EXISTING_PROPERTY,
    defaultImpl = SettlementAccountAttributeValueChangeEvent.class
)
public class SettlementAccountAttributeValueChangeEvent extends QuoteInformationRequiredEventBase implements ISettlementAccountAttributeValueChangeEvent {

  /**
   * The event data structure.
   */
  private @Valid SettlementAccountAttributeValueChangeEventPayload payload;
}